airflow/providers/amazon/aws/executors/ecs/README.md

```diff
@@ -0,0 +1,196 @@
+
+#### Mounting a Volume
+
+You can optionally mount a local directory as a volume on the container during run-time. This will allow you to make change to files on the mounted directory, and have those changes be reflected in the container. To do this, run the following command:
+
+```
+docker run --volume /abs/path/to/local/dir:/abs/path/to/remote/dir <image_name>
+```
+
+Note: Doing this will overwrite the contents of the directory on the container with the contents of the local directory.
+
+### Installing Python Dependencies
+
+This Dockerfile supports installing Python dependencies via `pip` from a `requirements.txt` file. Place your `requirements.txt` file in the same directory as the Dockerfile. If it is in a different location, it can be specified using the `requirements_path` build-argument. Keep in mind the Docker context when copying the `requirements.txt` file. Uncomment the two appropriate lines in the Dockerfile that copy the `requirements.txt` file to the container, and run `pip install` to install the dependencies on the container.
+
+### Building Image for ECS Executor
+
+Detailed instructions on how to use the Docker image, that you have created via this readme, with the ECS Executor can be found [here](link_to_how_to_guide).
+
+
+## Logging
+
+Airflow tasks executed via this executor run in ECS containers within the configured VPC. This means that logs are not directly accessible to the Airflow Webserver and when containers are stopped, after task completion, the logs would be permanently lost.
+
+Remote logging can be employed when using the ECS executor to persist your Airflow Task logs and make them viewable from the Airflow Webserver.
+
+### Configuring Remote Logging
+
+There are many ways to configure remote logging and several supported destinations. A general overview of Airflow Task logging can be found [here](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/logging-tasks.html). Instructions for configuring S3 remote logging can be found [here](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/logging/s3-task-handler.html) and Cloudwatch remote logging [here](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/logging/cloud-watch-task-handlers.html).
+Some important things to point out for remote logging in the context of the ECS executor:
+
+ - The configuration options for Airflow remote logging must be configured on the host running the Airflow Webserver (so that it can fetch logs from the remote location) as well as within the ECS container running the Airflow Tasks (so that it can upload the logs to the remote location). See [here](https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html) to read more about how to set Airflow configuration via config file or environment variable exports.
```

Comment:
Config should be consistent everywhere, not sure why we need to call this out explicity. See this in the [config ref](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html):

> Use the same configuration across all the Airflow components.

Reply:
I suppose it's similar to your suggestion of using the same airflow version, seems obvious but we had people hit this issue in UAT, which means folks are likely to hit it themselves out in the wild.

It is trickier than I think you're giving it credit to as well. It's easy to miss that the container must also have this config as well as the webserver.

Reply:
Let me rephrase. I don't care that we call out logging config needs to be consistent (it does!), but our focus should be on pointing out that _all_ config needs to be consistent (part of which is logging config).

To be honest, this config stuff is the part that I think needs the most attention before we merge this.

Reply:
In general, we are trying not to be too prescriptive. This isn't a readme on best practices for managing airflow, rather the need-to-knows for setting up this executor. All Airflow best practices go without saying IMHO.
BUT, on the other hand I'm also more than happy to update the language to note that config should be the same everywhere. Though, I'd still very much like to keep this section on logging config specifically.

Reply:
My concern is that if we feel the need to call out logging has to be consistent, that means the audience likely doesn't know everything needs to be consistent in the first place. It implies that you just need to worry about the couple things called out. Logging config isn't special in any way with this executor - it's just the 2nd or 3rd thing you notice if you haven't done it right (unless I'm overlooking something).